it/controllers/benefits/CompanyVanFuelBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,429 @@
+ * limitations under the License.
+ */
+
+/*
+  * Copyright 2021 HM Revenue & Customs
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *     http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+  package controllers.benefits
+
+  import controllers.employment.routes.CheckYourBenefitsController
+  import forms.YesNoForm
+  import models.User
+  import models.employment.{BenefitsViewModel, CarVanFuelModel}
+  import models.mongo.{EmploymentCYAModel, EmploymentDetails, EmploymentUserData}
+  import org.jsoup.Jsoup
+  import org.jsoup.nodes.Document
+  import play.api.http.HeaderNames
+  import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+  import play.api.libs.ws.WSResponse
+  import utils.{EmploymentDatabaseHelper, IntegrationTest, ViewHelpers}
+
+  class CompanyVanFuelBenefitsControllerISpec extends IntegrationTest with ViewHelpers with EmploymentDatabaseHelper {
+
+    val taxYearEOY: Int = taxYear-1
+    val employmentId: String = "001"
+
+    private val userRequest = User(mtditid, None, nino, sessionId, affinityGroup)(fakeRequest)
+
+    private def employmentUserData(isPrior: Boolean, employmentCyaModel: EmploymentCYAModel): EmploymentUserData =
+      EmploymentUserData(sessionId, mtditid, nino, taxYearEOY, employmentId, isPriorSubmission = isPrior, employmentCyaModel)
+
+    //def cyaModel(employerName: String, hmrc: Boolean, benefits: Option[BenefitsViewModel] = None): EmploymentCYAModel =
+    // EmploymentCYAModel(EmploymentDetails(employerName, currentDataIsHmrcHeld = hmrc), benefits)
+
+    def fullCarVanFuelModel: CarVanFuelModel =
+      CarVanFuelModel(
+        carVanFuelQuestion = Some(true),
+        carQuestion = Some(true),
+        car = Some(100),
+        carFuelQuestion = Some(true),
+        carFuel = Some(200),
+        vanQuestion = Some(true),
+        van = Some(300),
+        vanFuelQuestion = Some(true),
+        vanFuel = Some(400)
+      )
+
+    // models for Incomplete sections redirect tests
+    val benefitsWithNoBenefitsReceived: Option[BenefitsViewModel] = Some(BenefitsViewModel(isUsingCustomerData = true))
+
+    def benefitsWithNoCarVanFuelQuestion(carVanFuelQuestion: Option[Boolean] = Some(false)): Option[BenefitsViewModel] =
+      Some(BenefitsViewModel(isBenefitsReceived = true,
+        carVanFuelModel = Some(fullCarVanFuelModel.copy(carVanFuelQuestion = carVanFuelQuestion)),
+        isUsingCustomerData = true))
+
+    val benefitsWithEmptyCarQuestion: Option[BenefitsViewModel] =
```

Comment:
This isn't used and can be removed
(if you take the latest code I shared I already tidies these up)